### ISSUE TITLE
Array.squeeze()

### ISSUE THREAD

**Issue by barakugav** [rust-ndarray/ndarray#1395]
Is there a reason why `squeeze()` is not provided by the library? Its very useful.
I think it can be implemented fairly easily:
```Rust
pub trait Squeeze {
    fn squeeze(self) -> Self;
}
impl<S> Squeeze for ArrayBase<S, IxDyn>
where
    S: RawData,
{
    fn squeeze(self) -> Self {
        let mut out = self;
        for axis in (0..out.shape().len()).rev() {
            if out.shape()[axis] == 1 && out.shape().len() > 1 {
                out = out.remove_axis(Axis(axis));
            }
        }
        out
    }
}
```
Is there already such a function with a different name and i missed it?
Is there a problem with the implementation above?
Thanks for the support, this library is amazing!

**Comment by nilgoyette** [comment]
I don't think we have such a function in ndarray. I might have missed it too :) but in that case, we should at least add the word "squeeze" in the documentation!

I think your implementation above is right. Do not hesitate to submit a PR.

**Comment by barakugav** [comment]
Created a pull request: https://github.com/rust-ndarray/ndarray/pull/1396

**Comment by bluss** [comment]
Solved in #1396, shipped in 0.16.0
